lore/pipelines/holdout.py

```diff
@@ -197,7 +201,7 @@ def encode_x(self, data):
         dataframe = pandas.DataFrame(encoded)
         if self.index:
             dataframe.set_index(self.index)
-        return dataframe
+        return self.finalize(dataframe)
```

Comment:
The way I'd handle this without `finalize` would be to override `encode_x` in the child class, and have it perform whatever finalize would do after calling `super(...), without adding more methods to the API.